src/transformState.js

```diff
@@ -4,8 +4,27 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const item of actions) {
+    for (const key in item) {
```

Comment:
Sorry, I don`t understand why are you mapping all keys in action.. You have to catch every type of action and make changes with object. Example:
```
for (const { type, extraData, ...otherPropsYouNeed } of actions) {

switch (type) {
   case "extraData":
      ......your changes
      break;
   case "keysToRemove"
   ....... so on
}
}   
```